include/morphio/properties.h

```diff
@@ -206,59 +194,42 @@ struct Properties {
 std::ostream& operator<<(std::ostream& os, const PointLevel& pointLevel);
 
-template <>
-const std::vector<Point::Type>& Properties::get<Point>() const noexcept;
-template <>
-std::vector<Point::Type>& Properties::get<Point>() noexcept;
-
-template <>
-const std::vector<Perimeter::Type>& Properties::get<Perimeter>() const noexcept;
-template <>
-std::vector<Perimeter::Type>& Properties::get<Perimeter>() noexcept;
-
-template <>
-const std::vector<Diameter::Type>& Properties::get<Diameter>() const noexcept;
-template <>
-std::vector<Diameter::Type>& Properties::get<Diameter>() noexcept;
-
-template <>
-const std::vector<MitoSection::Type>& Properties::get<MitoSection>() const noexcept;
-template <>
-std::vector<MitoSection::Type>& Properties::get<MitoSection>() noexcept;
-
-template <>
-const std::vector<MitoPathLength::Type>& Properties::get<MitoPathLength>() const noexcept;
-template <>
-std::vector<MitoPathLength::Type>& Properties::get<MitoPathLength>() noexcept;
+#define INSTANTIATE_TEMPLATE_GET(T, M)                                       \
+    template <>                                                              \
+    inline std::vector<T::Type>& Properties::get_mut<T>() noexcept {         \
+        return M;                                                            \
+    }                                                                        \
+    template <>                                                              \
+    inline const std::vector<T::Type>& Properties::get<T>() const noexcept { \
+        return M;                                                            \
+    }
 
-template <>
-const std::vector<MitoNeuriteSectionId::Type>& Properties::get<MitoNeuriteSectionId>() const
-    noexcept;
-template <>
-std::vector<MitoNeuriteSectionId::Type>& Properties::get<MitoNeuriteSectionId>() noexcept;
+INSTANTIATE_TEMPLATE_GET(Point, _pointLevel._points)
+INSTANTIATE_TEMPLATE_GET(Perimeter, _pointLevel._perimeters)
+INSTANTIATE_TEMPLATE_GET(Diameter, _pointLevel._diameters)
+INSTANTIATE_TEMPLATE_GET(MitoSection, _mitochondriaSectionLevel._sections)
+INSTANTIATE_TEMPLATE_GET(MitoPathLength, _mitochondriaPointLevel._relativePathLengths)
+INSTANTIATE_TEMPLATE_GET(MitoNeuriteSectionId, _mitochondriaPointLevel._sectionIds)
+INSTANTIATE_TEMPLATE_GET(MitoDiameter, _mitochondriaPointLevel._diameters)
+INSTANTIATE_TEMPLATE_GET(Section, _sectionLevel._sections)
+INSTANTIATE_TEMPLATE_GET(SectionType, _sectionLevel._sectionTypes)
 
-template <>
-const std::vector<MitoDiameter::Type>& Properties::get<MitoDiameter>() const noexcept;
-template <>
-std::vector<MitoDiameter::Type>& Properties::get<MitoDiameter>() noexcept;
+#undef INSTANTIATE_TEMPLATE_GET
 
 template <>
-const std::vector<Section::Type>& Properties::get<Section>() const noexcept;
-template <>
-std::vector<Section::Type>& Properties::get<Section>() noexcept;
+inline const std::map<int32_t, std::vector<uint32_t>>& Properties::children<Section>() const
+    noexcept {
```

Comment:
What kind of line wrapping is this? Readability suffers a bit, I was wondering at first what a `noexcept` block is…

Reply:
clang-format